event/monitoring.go

```diff
@@ -22,6 +22,9 @@ type CommandStartedEvent struct {
 	CommandName  string
 	RequestID    int64
 	ConnectionID string
+	// ServerConnectionID contains the connection ID from the server of the operation. If the server does not return
+	// this value (e.g. on MDB < 4.2), it is unset.
+	ServerConnectionID *uint32
```

Comment:
Should this be an `*int32` to match the spec? I assume the server does not return a negative value for `serverConnectionID`. But I do not know how long running servers handle signed integer overflow.

Since this is public API, I suggest erring to the side of the spec. E.g. a future spec change could require drivers have `-1` mean something special.

Reply:
Hypothetically, the server should only return `connectionId` > 0. A `connectionId` <= 0 on the server represents an "internal" connection (and not an external one to a client). That said, the spec does describe the field as a signed integer, so I'll switch to `Int32`.